source/codegen/metadata_validation.py

```diff
@@ -321,6 +321,8 @@ def _validate_enum(enum_name: str, used_enums: Set[str], metadata: dict):
                     metadata, RULES.ENUMS_SHOULD_NOT_HAVE_DUPLICATE_VALUES, ["enums", enum_name]
                 ):
                     raise Exception(f"Duplicate values in enum!")
+        elif not enum.get("force-include", False):
+            raise Exception(f"Enum is in metadata but not referenced by function.")
```

Comment:
Maybe make this just a warning if `metadata["config"]["code_readiness"] != "Release"` ? That way we're not forced to comply with this rule while still in development.

Reply:
Yeah, good idea. I think it makes sense to not enforce it until a driver's set to release.